arrows/core/applets/render_mesh.cxx

```diff
@@ -129,78 +119,84 @@ stretch_to_byte_image(kwiver::vital::image_of<T> const& in_image)
 // ----------------------------------------------------------------------------
 void
 render_mesh::
-usage( std::ostream& outstream ) const
+add_command_options()
 {
-  outstream << "This tool renders a mesh into a depth or height image\n"
-            << "\n"
-            << "Usage: kwiver " << applet_name() << " [options] mesh camera image\n"
-            << "\n"
-            << "Options are:\b"
-            << "  -h | --help            displays usage information\n"
-            << "  --config | -c  FILE    Configuration for tool\n"
-            << "  --output-config  FILE  Dump configuration to file\n"
+  m_cmd_options->custom_help( wrap_text(
+       "This tool renders a mesh into a depth or height image\n"
+       "\n"
+       "Usage: kwiver " + applet_name() + " [options] mesh camera image"
+          ) );
+
+  m_cmd_options->positional_help( "\n   mesh - Mesh file name.\n"
+                                  "   camera - camera file name.\n"
+                                  "   output - Output image file name");
+
+  m_cmd_options->add_options()
+    ( "h,help",        "Display usage information" )
+    ( "c",             "Configuration file for tool" )
+    ( "output-config", "Dump configuration for tool", cxxopts::value<std::string>() )
+    ( "x",             "Output image width", cxxopts::value<int>()->default_value( "1920" ) )
+    ( "y",             "Output image height", cxxopts::value<int>()->default_value( "1080" ) )
+    ( "height-map",    "Render a height map instead of a depth map" )
+    ( "byte",          "Render as a byte image with scaled range" )
+
+    // positional parameters
+    ( "mesh-file",    "Mesh file name", cxxopts::value<std::string>() )
+    ( "camera-file",  "Camera file name", cxxopts::value<std::string>() )
+    ( "output-image", "Output image file name", cxxopts::value<std::string>() )
     ;
+
+    m_cmd_options->parse_positional({"mesh-file", "camera-file", "output-image", "extra"});
```

Comment:
What is "extra" here?  I see all the tools have it now, but I don't see where it is used.

Reply:
It would be a place to look for extra positional parameters beyond what precedes it in the list. It can be removed.